### PR TITLE
feat: Unit test for SD-JWT Flow (Issuer, Holder, Verifier)

### LINKED ISSUE #3461: Unit Test for SD-JWT Flow (Issuer, Holder, Verifier)
Add unit-test for SD-JWT Flow.

### PR DESCRIPTION
**Pull request by sandrask**
Unit test for SD-JWT Flow.

Closes #3461

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

**Review by sudeshrshetty — COMMENTED** [review]
From [Section 5.5.1](https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-02.html#section-5.5.1) of spec

> If no Holder Binding JWT is included, the Combined Format for Presentation ends with the ~ character after the last Disclosure.

It means in case of holder disclosure, there will be a stray `~` at the end of disclosures list

File: pkg/doc/sdjwt/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	t.Run("success", func(t *testing.T) {
+		// Issuer will issue SD-JWT for specified claims.
+		token, e := issuer.New(testIssuer, claims, nil, signer)
+		r.NoError(e)
+
+		// TODO: Should we have one call instead of two (designed based on JWT)
+		sdJWTSerialized, e := token.Serialize(false)
+		r.NoError(e)
+
+		fmt.Println(fmt.Sprintf("issuer SD-JWT: %s", sdJWTSerialized))
```
Comment (sudeshrshetty) [inline pkg/doc/sdjwt/integration_test.go:51]: is this `fmt.println()` leftover?

Comment (sandrask) [reply pkg/doc/sdjwt/integration_test.go:51]: No, I left print outs on purpose to show output formats.

**Comment by sandrask** [comment]
Yes, according to spec if there is no holder binding it should end with ~. I will fix this today since we currently don't have holder binding. I will work on holder binding next week.

**Pull request merged**